network/p2p/libp2pNode.go

```diff
@@ -81,66 +101,70 @@ type Node struct {
 	connMgr              *ConnManager
 }
 
-func NewLibP2PNode(logger zerolog.Logger,
-	id flow.Identifier,
-	address string,
-	conMgr *ConnManager,
-	key fcrypto.PrivateKey,
-	allowList bool,
-	rootBlockID string,
-	pingInfoProvider PingInfoProvider,
-	psOption ...pubsub.Option) (*Node, error) {
+type NodeOption func(node *Node) error
 
-	libp2pKey, err := PrivKey(key)
-	if err != nil {
-		return nil, fmt.Errorf("could not generate libp2p key: %w", err)
+// WithConnectionManager returns a NodeOption that sets the connection gater for the node
+func WithConnectionGator(gater *connGater) NodeOption {
+	return func(node *Node) error {
+		node.connGater = gater
+		return nil
 	}
+}
 
-	flowLibP2PProtocolID := generateFlowProtocolID(rootBlockID)
-
-	ctx, cancel := context.WithCancel(context.Background())
-
-	libP2PHost, connGater, pubSub, err := bootstrapLibP2PHost(ctx,
-		logger,
-		address,
-		conMgr,
-		libp2pKey,
-		allowList,
-		psOption...)
+// WithConnectionManager returns a NodeOption that sets the connection manager for the node
+func WithConnectionManager(connMgr *ConnManager) NodeOption {
+	return func(node *Node) error {
+		node.connMgr = connMgr
+		return nil
+	}
+}
 
-	if err != nil {
-		cancel()
-		return nil, fmt.Errorf("could not bootstrap libp2p host: %w", err)
+// WithPingService returns a NodeOption that initializes the PingService object for the node
+func WithPingService(rootBlockID string, pingInfoProvider PingInfoProvider) NodeOption {
+	return func(node *Node) error {
+		pingLibP2PProtocolID := generatePingProtcolID(rootBlockID)
+		pingService := NewPingService(node.host, pingLibP2PProtocolID, pingInfoProvider, node.logger)
+		node.pingService = pingService
+		return nil
 	}
+}
 
-	pingLibP2PProtocolID := generatePingProtcolID(rootBlockID)
-	pingService := NewPingService(libP2PHost, pingLibP2PProtocolID, pingInfoProvider, logger)
+// NewLibP2PNode creates a NewLibP2PNode using the given libp2p host and pubsub instances
+func NewLibP2PNode(id flow.Identifier,
```

Comment:
`NewLibP2PNode now expects all the mandatory params - `id`, `rootblock`, `libp2p host`, `pubsub` etc. to be passed in as without those, the node will not function. The node can be additionally be customized with `NodeOptions`